### ISSUE TITLE
pdf invoice info/addresses are dynamically modified 

### ISSUE THREAD

**Issue by kintaro1981** [thirtybees/ThirtyBees#479]
thirty bees 1.0.4 (upgraded following the blog article)

I changed the invoice address of the customer AFTER the order was created and the PDF invoice displays the updated address.

I think that it needs to display the OLD address.
I think that invoice needs to be "STATIC", or at least with the possibility to be manually modified but not dynamically modified with updated customer info.

**Comment by firstred** [comment]
True. I am always surprised when old invoices change. What would you suggest? Lock an address once there are one or more orders with that address?

**Comment by slvrspiral** [comment]
Magento does this by capturing the invoice and shipping address in a different table the can be updated but is done by the admins.  I think this is a good solution and am toying with doing it here too.

**Comment by Traumflug** [comment]
A pretty failsafe solution would be to store the created PDF. More in general, all documents sent to the customer.

This should also meet legal archiving requirements. Changing such documents after the fact is a no-go, even when done by admins.

**Comment by slvrspiral** [comment]
That may work for your purposes, but my company had different requirements.

**Comment by Dh42** [comment]
Some has changed and we are grabbing it from the wrong spot now. in the _order_invoice table the addresses are taken at the time of the order and stored in html format for the invoices. That is where we should be grabbing them from.

**Comment by kintaro1981** [comment]
is this fixed?
